src/scene_graph/scene_graph.hpp

```diff
@@ -8,6 +8,8 @@
 #include "../util/defines.hpp"
 #include "../resource_pool.hpp"
 
+#define MAX_INSTANCE_COUNT 768
```

Comment:
Prefer a `static const constexpr` in `d3d12_settings.hpp` over a definition

Reply:
That would require scene_graph.hpp to include d3d12_settings.hpp though